packages/compass-settings/src/components/settings/theme.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback } from 'react';
+import { connect } from 'react-redux';
+import {
+  Body,
+  Checkbox,
+  Label,
+  Description,
+  css,
+  spacing,
+  RadioBoxGroup,
+  RadioBox,
+} from '@mongodb-js/compass-components';
+import type { RootState } from '../../stores';
+import { changeFieldValue } from '../../stores/updated-fields';
+import type {
+  PreferenceStateInformation,
+  THEMES,
+} from 'compass-preferences-model';
+import { settingStateLabels } from './state-labels';
+
+type ThemeSettingsProps = {
+  handleChange: (field: 'theme', value: THEMES) => void;
+  preferenceStates: PreferenceStateInformation;
+  themeValue: THEMES;
+};
+
+const checkboxStyles = css({
+  marginTop: spacing[3],
+  marginBottom: spacing[3],
+});
+
+export const ThemeSettings: React.FunctionComponent<ThemeSettingsProps> = ({
+  themeValue,
+  preferenceStates,
+  handleChange,
+}) => {
+  const handleOSCheckboxChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      handleChange('theme', event.target.checked ? 'OS_THEME' : 'LIGHT');
+    },
+    [handleChange]
+  );
+  const handleSelectorChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      handleChange('theme', event.target.value as THEMES);
+    },
+    [handleChange]
+  );
+
+  return (
+    <div data-testid="theme-settings">
+      <Body>Change the appearance of Compass.</Body>
+
+      <div>
+        <Checkbox
+          className={checkboxStyles}
+          name="use-os-theme"
+          id="use-os-theme"
+          data-testid="use-os-theme"
+          label={
+            <>
+              <Label htmlFor="use-os-theme">Sync with OS</Label>
+              <Description>
+                Automatically switch between light and dark themes based on your
+                OS settings
+              </Description>
+            </>
+          }
+          onChange={handleOSCheckboxChange}
+          checked={themeValue === 'OS_THEME'}
+          disabled={!!preferenceStates.theme}
+        />
+        {settingStateLabels[preferenceStates.theme ?? '']}
+      </div>
+
+      <RadioBoxGroup
+        id="theme-selector"
+        onChange={handleSelectorChange}
+        value={themeValue}
+      >
+        <RadioBox
+          id="theme-selector-light"
+          data-testid="theme-selector-light"
+          value="LIGHT"
+          disabled={!!preferenceStates.theme || themeValue === 'OS_THEME'}
+        >
+          Light Theme
+        </RadioBox>
+        <RadioBox
+          id="theme-selector-dark"
+          data-testid="theme-selector-dark"
+          value="DARK"
+          disabled={!!preferenceStates.theme || themeValue === 'OS_THEME'}
+        >
+          Dark Theme
```

Comment:
This should ideally look more like https://www.figma.com/file/bSnDxjrKFf9EROX7X6FJoB/Compass-Nav-tree%2BSettings?node-id=25%3A3098, my Figma skills just aren't good enough to extract the relevant parts as SVG.

Reply:
Thanks to @gribnoysup the SVGs are included now :slightly_smiling_face: There’s a slight bit of awkwardness in that the darkreader dark mode turns the light mode icon dark when it’s on, but that might be okay, esp. since it’s not a forever solution (hopefully):

LG dark mode:
![Screenshot from 2022-10-26 17-02-33](https://user-images.githubusercontent.com/899444/198066233-87590f32-a314-49c0-b79e-851c0105ea99.png)

Darkreader:
![Screenshot from 2022-10-26 17-01-40](https://user-images.githubusercontent.com/899444/198066246-931d2820-f52b-4772-be86-c535e213a201.png)

(and sorry for the inconsistent cropping)

Reply:
Oh, darkreader changing our light icon to the dark one is very meta 😆 Depending on how much time you feel like spending on this, you can probably convert this svg to data-url and use an img tag instead of svg directly, I would hope that in this case darkreader will not alter the content.

But also I agree that it's probably fine for now